src/InsertCapablePdoTrait.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use ArrayAccess;
+use Dhii\Util\String\StringableInterface;
+use OutOfRangeException;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use InvalidArgumentException;
+use PDOStatement;
+use Psr\Container\ContainerInterface;
+use stdClass;
+use Traversable;
+
+/**
+ * Common functionality for objects that can insert records into a database using PDO.
+ *
+ * @since [*next-version*]
+ */
+trait InsertCapablePdoTrait
+{
+    /**
+     * Executes an INSERT SQL query, inserting records into the database.
+     *
+     * @since [*next-version*]
+     *
+     * @param array[]|ArrayAccess[]|stdClass[]|ContainerInterface[]|Traversable $records A list of records to insert.
+     *
+     * @throws ContainerExceptionInterface If an error occurred while reading from a record's container.
+     *
+     * @return PDOStatement The executed PDO statement.
+     */
+    protected function _insert($records)
+    {
+        $processedRecords = $this->_preProcessRecords($records, $valueHashMap);
+
+        $query = $this->_buildInsertSql(
+            $this->_getSqlInsertTable(),
```

Comment:
What happens if an entity spans multiple tables?

Reply:
Don't know. Cannot insert into multiple tables in a single SQL statement. So that's probably responsibility that should exist at a higher level of abstraction.

What I would probably do is have a composite resource model, that understands what the entity is made of, split them accordingly and delegate separate queries to the different children resource models.

That's just what I would do. In any case, this is a trait for providing generic PDO insertion functionality. And insertions cannot handle multiple tables.

Reply:
It's true that insertion cannot be done to multiple tables in one query. But whether one or many - that can be abstracted. We treat CQRS resource models as something that is aware of an entity. An entity may span multiple tables. If it can be retrieved from multiple tables, then shouldn't it be also insertable to multiple tables, etc? Otherwise, how do you retrieve a multi-table entity, and then make a change to it?

Reply:
I agree with you. But this trait is not a trait for a resource model. It's just functionality for inserting records into the database. A higher level trait or class should handle something like this. Do you not agree?